src/gc/proxy.d

```diff
@@ -28,32 +28,67 @@ private
     extern (C) void thread_init();
     extern (C) void thread_term();
 
+    import core.internal.spinlock;
+    static SpinLock instanceLock;
+
     __gshared GC instance;
     __gshared GC proxiedGC; // used to iterate roots of Windows DLLs
-
 }
 
+alias GCInitNoThrow = void function() nothrow @nogc;
 
 extern (C)
 {
 
     void gc_init()
     {
-        config.initialize();
-        ManualGC.initialize(instance);
-        ConservativeGC.initialize(instance);
-        if (instance is null)
-        {
-            import core.stdc.stdio : fprintf, stderr;
-            import core.stdc.stdlib : exit;
+        import core.atomic : atomicLoad, atomicStore;
+
+        auto pinstance = cast(shared(GC)*) &instance;
 
-            fprintf(stderr, "No GC was initialized, please recheck the name of the selected GC ('%.*s').\n", cast(int)config.gc.length, config.gc.ptr);
-            exit(1);
+        if (atomicLoad(*pinstance) is null)
```

Comment:
whenever I do double-checked locking, I like to put a comment :)
`// using double-checked locking`